pkg/backup/config.go

```diff
@@ -0,0 +1,44 @@
+package backup
+
+import (
+	"fmt"
+	"github.com/k0sproject/k0s/internal/util"
+	"github.com/sirupsen/logrus"
+	"os"
+	"path"
+)
+
+type configurationStep struct {
+	path string
+}
+
+func newConfigurationStep(path string) *configurationStep {
+	return &configurationStep{path: path}
+}
+
+func (c configurationStep) Name() string {
+	return "k0s.yaml"
+}
+
+func (c configurationStep) Backup() (StepResult, error) {
+	_, err := os.Stat(c.path)
+	if os.IsNotExist(err) {
+		logrus.Info("default k0s.yaml is used, do not back it up")
+		return StepResult{}, nil
+	}
+	if err != nil {
+		return StepResult{}, fmt.Errorf("can't backup `%s`: %v", c.path, err)
+	}
+	return StepResult{filesForBackup: []string{c.path}}, nil
+}
+
+func (c configurationStep) Restore(restoreFrom, restoreTo string) error {
+	if !util.FileExists(restoreFrom) {
+		logrus.Info("No k0s.yaml in the backup archive")
+		return nil
+	}
+	logrus.Info("Previously used k0s.yaml saved under the data directory")
+	objectPathInArchive := path.Join(restoreFrom, "k0s.yaml")
+	objectPathInRestored := path.Join(restoreTo, "k0s.yaml")
+	return util.FileCopy(objectPathInArchive, objectPathInRestored)
```

Comment:
so the general flow here is to first unpack archive to temporary directory and then cherrypick everything defined from that temporary directory. 

If the question is generic like "why do we want k0s.yaml in backup", the answer is because we have some helm related setting there that can affect the cluster state pretty significantly.